common/regression_test.py

```diff
@@ -0,0 +1,40 @@
+import magma as m
+
+
+def parse_genesis_circuit(circuit):
```

Comment:
not sure if this is a general thing we should design for...unfortunately there's no better way to do this since Genesis flattens all inputs (no array types). Probably fine for now but something better than pattern matching on names is warranted.